src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,119 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 1 << 4;
+    private static final int MAX_CAPACITY = 1 << 30;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(hash(key), key, value, null);
+        if (table == null) {
+            table = new Node[INITIAL_CAPACITY];
+        }
+        int i = index(hash(key));
+        Node<K, V> currentNode = table[i];
+        int threshold = (int) (table.length * LOAD_FACTOR);
+        if (currentNode == null) {
+            table[i] = newNode;
+            if (++size > threshold) {
+                resize();
+            }
+            return;
+        }
+        do {
+            if (keyCheck(key, currentNode)) {
+                currentNode.value = value;
+                return;
+            }
+            if (currentNode.next == null) {
+                currentNode.next = newNode;
+                if (++size > threshold) {
+                    resize();
+                }
+                return;
+            }
+            currentNode = currentNode.next;
+        } while (true);
     }
 
     @Override
     public V getValue(K key) {
+        if (table != null) {
+            for (Node<K, V> currentNode : table) {
+                if (currentNode != null) {
+                    do {
+                        if (keyCheck(key, currentNode)) {
+                            return currentNode.value;
+                        }
+                        currentNode = currentNode.next;
+                    } while (currentNode != null);
+                }
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        private Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private int hash(Object key) {
+        int hash;
+        return (key == null) ? 0 : (hash = key.hashCode()) ^ (hash >>> 16);
+    }
+
+    private int index(int hash) {
```

Comment:
methods names should starts from verb https://mate-academy.github.io/style-guides/java/java.html#s5.2.3-method-names